File: client/src/app/core/models/order.ts
import { Customer } from "./customer";
import { Delivery } from "./delivery";
import { OrderProduct } from "./orderProduct";
import { OrderStatus } from "./orderStatus";

export interface Order {
  id: number;
  createdAt: Date;
  customer: Customer;
  shippingAddressSameAsCustomer: boolean;
  status: OrderStatus;
  products : OrderProduct[];
  delivery : Delivery;
}

File: api/Entities/Order.cs
using System;

namespace api.Entities
{
    public class Order
    {
        public int Id { get; set; }
        public DateTime CreatedAt  { get; set; }
        public int CustomerId { get; set; }
        public Customer Customer { get; set; }
        public int DeliveryId { get; set; }
        public Delivery Delivery { get; set; }
        public bool ShippingAddressSameAsCustomer { get; set; }
        public OrderStatus Status { get; set; }
        public float DeliveryPrice  { get; set; }
    }
}